pkg/signature/ecdsa.go

```diff
@@ -181,6 +185,11 @@ func (e ECDSAVerifier) VerifySignature(signature, message io.Reader, opts ...Ver
 		return fmt.Errorf("reading signature: %w", err)
 	}
 
+	// Without this check, VerifyASN1 panics on an invalid curve.
+	if !e.publicKey.Curve.IsOnCurve(e.publicKey.X, e.publicKey.Y) {
+		return fmt.Errorf("invalid ECDSA curve for %s", e.publicKey.Params().Name)
```

Comment:
small nit: the ecdsa public key (point) is invalid (not on the given ECDSA curve)

Reply:
Thanks! Updated.